src/Validation/Validator.php

```diff
@@ -295,8 +295,24 @@ private function validateFiles(array $files, array $validationReference, string
 
 			// Loop all validations from the reference.
 			foreach ($reference as $dataKey => $dataValue) {
-				// Check validation for accept file types.
+				// Check validation for accepted file types.
 				if ($dataKey === 'accept') {
+					$individualFiles = [];
+					for ($i = 0; $i < count($fileValue['name']); $i++) {
+						$file = [
+							'name' => $fileValue['name'][$i],
+							'type' => $fileValue['type'][$i],
+							'tmp_name' => $fileValue['tmp_name'][$i],
+						];
+						$individualFiles[] = $file;
+					}
+
+					foreach ($individualFiles as $file) {
+						if (!$this->isMimeTypeValid($file)) {
+							$output[$fileKey] = sprintf($this->labels->getLabel('validationAcceptMime', $formId), $dataValue);
+						}
+					}
```

Comment:
This is true - `$file` remains in scope of the function, and isn't scoped to the `foreach` loop - but I'm not sure how I feel about your comment. I'm not _against_ it, but I also don't find myself agreeing completely.

The code here is assigning a new value to `$file` upon entering the `foreach` loop (and not consuming it in any way), and we generally take care to reuse variables where it makes sense.  Adding an `unset` would prevent consuming the variable in an unexpected place, but this argument could be used for any single variable we're declaring. It's probably a generally good idea to be more mindful of the ways in which we use and consume variables, but this is an iterator that's reinitialized on its next use. It doesn't seem like an actual concern to me - I don't see how it could cause problems not to unset it if we're _mindful_ ✨  of our variable use and don't consume it for some reason in a different place. Pushing it further, `unset` doesn't free memory, so unsetting it and then declaring a new iterator with the same name could lead to double the memory consumption here. That's probably isn't an actually substantial concern, however.

We also discussed using `reset`, which supposedly comes with performance benefits. I've done some benchmarking on 3v4l, and it seems that neither `unset` or `reset` result in better performance of foreach looping: https://3v4l.org/qfZSk, https://3v4l.org/6B5lg3, https://3v4l.org/EAgOP - test performance does differ between PHP versions (in various ways, which is surprising), but for _most_ PHP versions, it takes `4.0531158447266E-6` to foreach through a 100-element array, no matter whether using `reset`, both `unset` and `reset` or none at all. This is probably partially so because `foreach` itself resets the internal array pointer.

My reasoning regarding this is that I don't see how the benefit of making sure devs don't consume `$file` inbetween the two `foreach` loops outweighs code understandability (with `unset` and `reset` being called), and I don't see any other substantial reasons to use `unset` and none whatsoever to use `reset`. This isn't a hill I'm willing to die upon, though - and if anyone has different feelings regarding this, please let me know.